operator/v1/types_console.go

```diff
@@ -99,8 +99,51 @@ type StatuspageProvider struct {
 	PageID string `json:"pageID"`
 }
 
+// WidgetName defines names of UI widgets in the console UI.
+type WidgetName string
+
+const (
+	// lightspeedButton is the name for the Lightspeed button HTML element.
+	LightspeedButton WidgetName = "LightspeedButton"
+)
+
+// WidgetState defines the state of the widget in the console UI.
+type WidgetState string
+
+const (
+	// "Show" means that the widget will be rendered in the console UI.
+	WidgetShow WidgetState = "Show"
+	// "Hide" means that the widget will not be rendered in the console UI.
+	WidgetHide WidgetState = "Hide"
+)
+
+// WidgetVisibility defines the criteria to show/hide a widget.
+// +union
+type WidgetVisibility struct {
+	// state defines the if the widget is shown or hidden in the console UI.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="Show";"Hide"
+	// +kubebuilder:validation:Required
+	State WidgetState `json:"state"`
+}
+
+// Widgets contains set of UI widgets and their state in the console UI.
+type Widget struct {
+	// name is the unique name of an widget.
+	// +kubebuilder:validation:Enum:="LightspeedButton";
+	// +kubebuilder:validation:Required
+	Name WidgetName `json:"name"`
+	// visibility defines the visibility state of widget.
+	// +kubebuilder:validation:Required
+	Visibility WidgetVisibility `json:"visibility"`
+}
+
 // ConsoleCustomization defines a list of optional configuration for the console UI.
 type ConsoleCustomization struct {
+	// widgets defines an array of widgets that can be interacted with in the console UI.
+	// Each widget defines a visual state, by which it will get rendered.
+	// +kubebuilder:validation:Required
```

Comment:
You can't add a new required field to an existing struct, how would this work for upgrades and existing clusters?

Reply:
true 👍